src/bigquery.ts

```diff
@@ -0,0 +1,1177 @@
+  routineClient?: RoutineServiceClient;
+  rowaccesspolicyClient?: RowAccessPolicyServiceClient;
+};
+
+export class BigQueryClient {
+  datasetClient: DatasetServiceClient;
+  tableClient: TableServiceClient;
+  jobClient: JobServiceClient;
+  modelClient: ModelServiceClient;
+  routineClient: RoutineServiceClient;
+  rowaccesspolicyClient: RowAccessPolicyServiceClient;
+
+  /**
+   * @param {object} [bigQueryClientOptions] - Enables user to instantiate clients separately and use those as the subclients.
+   * @param {object} [subClientOptions] - These options will be shared across subclients.
+   * To have sub-clients with different options, instantiate each client separately.
+   */
+  constructor(
+    options?: bigQueryClientOptions,
+    subClientOptions?: subClientOptions
+  ) {
+    this.datasetClient =
+      options?.datasetClient ??
+      new DatasetServiceClient(
+        subClientOptions?.opts,
+        subClientOptions?.gaxInstance
+      );
+    this.tableClient =
+      options?.tableClient ??
+      new TableServiceClient(
+        subClientOptions?.opts,
+        subClientOptions?.gaxInstance
+      );
+    this.jobClient =
+      options?.jobClient ??
+      new JobServiceClient(
+        subClientOptions?.opts,
+        subClientOptions?.gaxInstance
+      );
+    this.modelClient =
+      options?.modelClient ??
+      new ModelServiceClient(
+        subClientOptions?.opts,
+        subClientOptions?.gaxInstance
+      );
+    this.routineClient =
+      options?.routineClient ??
+      new RoutineServiceClient(
+        subClientOptions?.opts,
+        subClientOptions?.gaxInstance
+      );
+    this.rowaccesspolicyClient =
+      options?.rowaccesspolicyClient ??
+      new RowAccessPolicyServiceClient(
+        subClientOptions?.opts,
+        subClientOptions?.gaxInstance
+      );
+  }
+  getDataset(
+    request?: protos.google.cloud.bigquery.v2.IGetDatasetRequest,
+    optionsOrCallback?:
+      | CallOptions
+      | Callback<
+          protos.google.cloud.bigquery.v2.IDataset,
+          protos.google.cloud.bigquery.v2.IGetDatasetRequest | null | undefined,
+          {} | null | undefined
+        >,
+    callback?: Callback<
+      protos.google.cloud.bigquery.v2.IDataset,
+      protos.google.cloud.bigquery.v2.IGetDatasetRequest | null | undefined,
+      {} | null | undefined
+    >
+  ): Promise<
+    [
+      protos.google.cloud.bigquery.v2.IDataset,
+      protos.google.cloud.bigquery.v2.IGetDatasetRequest | undefined,
+      {} | undefined,
+    ]
+  > | void {
+    request = request || {};
+    let options: CallOptions;
+    if (typeof optionsOrCallback === 'function' && callback === undefined) {
+      callback = optionsOrCallback;
+      options = {};
+    } else {
+      options = optionsOrCallback as CallOptions;
+    }
+    if (callback === undefined) {
+      return this.datasetClient.getDataset(request, options);
+    }
+    return this.datasetClient.getDataset(request, options, callback);
+  }
```

Comment:
should we copy the docstring/comment for each method ? Right now users will not have documentation for the methods on the uber client.

Reply:
I am remembering that the reason I didn't do this is because I could not figure out how 😅  do we want to 100% block on this? Or would you be okay if I opened a follow up issue? It's ok if you want to block - I can sink some more time in on Monday

Reply:
GOT IT. It was terrible. But I got it. 😁